cmd/root.go

```diff
@@ -157,15 +158,15 @@ func main(*cobra.Command, []string) error {
 		log.SetLevel(log.DebugLevel)
 	}
 
-	clusterSources, err := createSources(c.Clusters)
+	clusterSources, err := createSources(c, c.Clusters)
```

Comment:
nit: if giving the whole config here already, the second argument is a bit redundant as it can be accessed anyway in `createSources` via `c.Clusters`. 

If you want to be explicit I guess a `c.IngressNodeSelector, c.Clusters` here would make sense as well.